folium/features.py

```diff
@@ -787,6 +787,134 @@ def __init__(self, location, radius=500, color='black',
             """)
 
 
+class CustomLegend(Layer):
```

Comment:
@olibchr Maybe it is better if you send this in a separate PR. It will be easier to merge.